screenboards_test.go

```diff
@@ -1,95 +1,224 @@
+										},
+										{
+											Palette:    String("white_on_green"),
+											Comparator: String("<"),
+											Value:      nil,
+										},
+									},
+								},
+							},
+						},
+					},
+					{
+						Type: String("toplist"),
+						TileDef: &TileDef{
+							Viz: String("toplist"),
+							Requests: []TileDefRequest{
+								{
+									ConditionalFormats: []ConditionalFormat{
+										// number value in JSON
+										{
+											Palette:    String("white_on_red"),
+											Comparator: String(">"),
+											Value:      Strint("1"),
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			file: "screenboard_response_widget_float_position",
+			want: &Screenboard{
+				Widgets: []Widget{
+					{
+						Type: String("free_text"),
+						Text: String("processed"),
+						X:    Float32(1),
+						Y:    Float32(70.83333333333334),
+					},
+				},
+			},
+		},
+		{
+			file: "screenboard_response_manage_status_titlesize",
+			want: &Screenboard{
+				Widgets: []Widget{
+					{
+						Type:                  String("manage_status"),
+						ManageStatusTitleSize: Strint("16"),
+					},
+				},
+			},
+		},
 	}
-
-	for _, widget := range screenboard.Widgets {
-		validateWidget(t, widget)
-	}
-}
-
-func validateWidget(t *testing.T, wd Widget) {
-	expectedType := "image"
-	if widgetType := wd.GetType(); widgetType != expectedType {
-		t.Fatalf("expect type %s. Got %s", expectedType, widgetType)
-	}
-
-	expectedHeight := 20
-	if height := wd.GetHeight(); height != expectedHeight {
-		t.Fatalf("expect height %d. Got %d", expectedHeight, height)
-	}
-
-	expectedWidth := 32
-	if width := wd.GetWidth(); width != expectedWidth {
-		t.Fatalf("expect width %d. Got %d", expectedWidth, width)
-	}
-
-	expectedX := 32
-	if x := wd.GetX(); x != expectedX {
-		t.Fatalf("expect x %d. Got %d", expectedX, x)
-	}
-
-	expectedY := 7
-	if y := wd.GetY(); y != expectedY {
-		t.Fatalf("expect y %d. Got %d", expectedY, y)
-	}
-
-	expectedURL := "http://path/to/image.jpg"
-	if url := wd.GetURL(); url != expectedURL {
-		t.Fatalf("expect url %s. Got %s", expectedURL, url)
+	for _, tt := range tests {
+		t.Run(tt.file, func(t *testing.T) {
+			ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				response, err := ioutil.ReadFile("./testdata/fixtures/" + tt.file + ".json")
+				if err != nil {
+					// mustn't call t.Fatal from a different Goroutine
+					http.Error(w, err.Error(), http.StatusInternalServerError)
+					return
+				}
+				w.Write(response)
+			}))
+			defer ts.Close()
+
+			datadogClient := Client{
+				baseUrl:      ts.URL,
+				HttpClient:   ts.Client(),
+				RetryTimeout: 5 * time.Second,
+			}
+
+			got, err := datadogClient.GetScreenboard(6334)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if !reflect.DeepEqual(got, tt.want) {
+				// Go formatting doesn't handle pointers gracefully, so we'll display the diff as JSON
```

Comment:
Alternatively use `github.com/davecgh/go-spew` which is already vendored in and in use by `testify`, something like this would work (untested):

```golang
t.Errorf("got:\n%s\nwant:\n%s", spew.Sdump(got), spew.Sdump(tt.want))
```

Reply:
Ah, nice, I didn't want to add another dependency, but if there already is one, all the better!